src/Kestrel.Transport.Sockets/SocketConnection.cs

```diff
@@ -111,25 +133,36 @@ internal SocketConnection(Socket socket, SocketTransport transport)
             catch (SocketException ex) when (ex.SocketErrorCode == SocketError.ConnectionReset)
             {
                 error = new ConnectionResetException(ex.Message, ex);
+                _trace.ConnectionReset(ConnectionId);
             }
-            catch (SocketException ex) when (ex.SocketErrorCode == SocketError.OperationAborted)
+            catch (SocketException ex) when (ex.SocketErrorCode == SocketError.OperationAborted ||
+                                             ex.SocketErrorCode == SocketError.Interrupted)
             {
                 error = new ConnectionAbortedException();
+                _trace.ConnectionError(ConnectionId, error);
             }
             catch (ObjectDisposedException)
             {
                 error = new ConnectionAbortedException();
+                _trace.ConnectionError(ConnectionId, error);
             }
             catch (IOException ex)
             {
                 error = ex;
+                _trace.ConnectionError(ConnectionId, error);
             }
             catch (Exception ex)
             {
                 error = new IOException(ex.Message, ex);
+                _trace.ConnectionError(ConnectionId, error);
             }
             finally
             {
+                if (_aborted)
+                {
+                    error = error ?? new ConnectionAbortedException();
```

Comment:
Should we log in the ConnectionAbortedException case?

Reply:
Just because the error is being set here, doesn't mean that there is an ongoing call to `Request.Body.ReadAsync()` for the error to be thrown from. This error always gets set for a server-initiated connection close. If there is an ongoing `ReadAsync()` call when the server closes the connection, it's because the application called `Abort()`.

We log a disconnect in the `Abort()` call that sets `_aborted` flag [here](https://github.com/aspnet/KestrelHttpServer/blob/08c6c3866791334d22f6137d3a0548d42ba067c2/src/Kestrel.Core/Internal/Http/Http1OutputProducer.cs#L153). This covers both the libuv and sockets transports.

We could look to changing the log in Http1OutputProducer to make it more clear whether the connection was aborted or closed normally, though this is usually clear when the exception from `ReadAsync` bubbles up and gets logged.